### ISSUE TITLE
Disable CLIs setting value not persisting in GUI

### ISSUE THREAD

**Issue by AdamJNavarro** [withfig/fig#627]
### Description:

When updating the `Disable CLIs` setting, the values inserted do not persist in the GUI despite the functionality working as expected. Upon re-opening settings, you'll see the input is empty with the placeholder being displayed. In order to restore autocomplete for previously entered tools, I have to add a space value into the input.

https://user-images.githubusercontent.com/8992878/143081013-022062e1-d106-459a-9b6e-8411a8a23d68.mov

### Details:
|macOS|Fig|Shell|
|-|-|-|
|12.0.1|Version 1.0.52 (B317)|/bin/zsh|
<details><summary><code>fig diagnostic</code></summary>
<p>
<pre>Version 1.0.52 (B317) [U.S.] 
UserShell: /bin/zsh
Bundle path: /Applications/Fig.app
Autocomplete: true
Settings.json: true
CLI installed: true
CLI tool path: /Users/adamjnav/.fig/bin/fig
Accessibility: true
Number of specs: 7
SSH Integration: false
Tmux Integration: false
Keybindings path: <none>
iTerm Integration: true [Authenticated]
Hyper Integration: false
VSCode Integration: true
Docker Integration: false
Symlinked dotfiles: false
Only insert on tab: false
Installation Script: true
PseudoTerminal Path: 
SecureKeyboardInput: false
SecureKeyboardProcess: <none>
Current active process: ??? (???) - ???
Current working directory: ???
Current window identifier: ???</pre>
</p>
</details>

**Comment by mschrage** [comment]
Hey @AdamJNavarro! Thanks so much for reporting this bug. 

Can you update this setting in the GUI app, then run `fig settings autocomplete.disableForCommands` in a terminal and share the output here?

This will help narrow down whether the issue is related to the presentation in the settings app UI or whether the value is not actually being written to disk.

**Comment by AdamJNavarro** [comment]
Hey @mschrage, sorry for the delay in response. The command above correctly outputs the clis I've entered in the Settings UI but now it seems that it doesn't correctly disable the CLIs entered. As well, the Settings UI still is empty upon closing and re-opening.

Everything from my initial `fig diagnostic` report is the same except for I've since upgraded the version to `Version 1.0.53 (B339) [U.S.]`

<img width="631" alt="Screen Shot 2021-12-13 at 10 57 46 AM" src="https://user-images.githubusercontent.com/8992878/145855734-11c966c7-4c85-45d9-bafa-0bca54600cfe.png">
